### PR TITLE
Covering in Go unit tests

### PR DESCRIPTION
**Pull request by mvshao**
**Description**

Changes proposed in this pull request:

- generate cover for unit test using `go test -cover` command

**Related issue(s)**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,7 +56,7 @@ vet: ## Run go vet against code.
 
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test ./... -coverprofile cover.out
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test ./... -cover
```
Comment (VOID404) [inline Makefile:59]: How about making it a toggle with an env var?

```suggestion
ifdef COVERFILE
	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test ./... -coverprofile cover.out
else
	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test ./... -cover
endif
```

Comment (mvshao) [reply Makefile:59]: If you are using the `cover.out` file to check the coverage on a daily basis, I can commit your suggestion.

**Review by VOID404 — APPROVED** [review]

**Pull request merged**